File: tests/integration/construct_component_unknown/nodejs/component.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as pulumi from "@pulumi/pulumi";

interface ComponentArgs {
    message: pulumi.Input<string>;
    nested: pulumi.Input<{
        value: pulumi.Input<string>;
    }>;
}

export class Component extends pulumi.ComponentResource {
    constructor(name: string, args: ComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("testcomponent:index:Component", name, args, opts, true);
    }
}

File: tests/integration/construct_component_unknown/dotnet/Component.cs
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

using Pulumi;

class Component : Pulumi.ComponentResource
{
    public Component(string name, ComponentArgs args, ComponentResourceOptions? opts = null)
        : base("testcomponent:index:Component", name, args, opts, remote: true)
    {
    }
}

class ComponentArgs : Pulumi.ResourceArgs
{
    [Input("message")]
    public Input<string> Message { get; set; } = null!;
    [Input("nested")]
    public Input<ComponentNestedArgs> Nested { get; set; } = null!;
}

class ComponentNestedArgs : Pulumi.ResourceArgs
{
    [Input("value")]
    public Input<string> Value { get; set; } = null!;
}
